app/controllers/stories_controller.rb

```diff
@@ -241,8 +241,24 @@ def assign_feed_stories
       @stories = Articles::Feeds::Latest.call
     else
       @default_home_feed = true
-      feed = Articles::Feeds::LargeForemExperimental.new(page: @page, tag: params[:tag])
-      @featured_story, @stories = feed.featured_story_and_default_home_feed(user_signed_in: user_signed_in?)
+      strategy = AbExperiment.get(experiment: :feed_strategy, controller: self, user: current_user,
+                                  default_value: "original")
+      feed = if strategy.weighted_query_strategy?
+               # I'm uncertain why we don't pass a user here, but it mimics
+               # the behavior of the original LargeForemExperimental.
+               Articles::Feeds::WeightedQueryStrategy.new(user: nil, page: @page, tags: params[:tag])
+             else
+               Articles::Feeds::LargeForemExperimental.new(page: @page, tag: params[:tag])
+             end
+      Datadog.tracer.trace("#{self.class}.#{__method__}", span_type: "feed", resource: feed.class.to_s) do
```

Comment:
Amendment to my above comment: we may actually want to have a consistent operation name (the first parameter) so we can compare them side by side. Operation names are intended to be super low-cardinality. Here are the ones we currently have in place (defined by Datadog's Rails integration) for the Rails service:

<img width="213" alt="Datadog Rails operation names" src="https://user-images.githubusercontent.com/108205/140553385-dc6fd3ac-fb63-4362-ab6d-23e79498d93c.png">

The naming convention doesn't matter as much — these are just named that way because that's how `ActiveSupport::Instrumentation` defines them. Resources and spans can be super high-cardinality, but operation names being high-cardinality unfortunately makes it harder to compare and contrast them.

With that in mind, is there a name we could use that covers all of them, like `feed.query`? And if we do that, would it make sense to distinguish where it runs in the `resource`?

Reply:
I can go with `feed.query` and adjust the `resource`.

Reply:
@jgaskins see the latest change